Correct confusing headers in HLSLDocs

pow2clk opened the pull request:
AvailabilityDiagnostics.rst and ExpectedDifferences.rst both had multiple headers that were perceived to be the "top-level".

 In  AvailabilityDiagnostics.rst two headers had both over and underlines. The second was the "Examples" section so it showed up in the top level HLSL docs. The overline is removed here so it's clear it's a subheader.

In ExpectedDifferences.rst, the first header had no overline, so a few headers that looked the same were included. The overline is added to the top header to make clear that it's the main header.

Commit:
Correct confusing header in HLSLDocs

Because AvailabilityDiagnostics.rst mistakenly overlined the "Examples"
section, it was included in the generated HLSLDocs page.
By demoting it to a subheader, it shouldn't show up as a top-level
HLSLDocs page.

pow2clk commented:
See https://clang.llvm.org/docs/HLSL/HLSLDocs.html to see the effect

Commit:
Add another case with multiple top-level headers

ExpectedDifference sections were bleeding into the top page too

bogner commented on clang/docs/HLSL/AvailabilityDiagnostics.rst:
```diff
@@ -52,7 +52,6 @@ If the compilation target is a shader library, only availability based on shader
 
-========
 Examples
 ========
```
Should we switch the examples themselves to use `----` instead of `====` so that they stay nested under the "Examples" heading?

pow2clk replied:
Yes!

Commit:
make examples subjects of Examples again

Justin pointed out that the earlier change moved the examples out
from the examples section. By demoting them as well, they will stay put

hekota commented:
Thanks @pow2clk !
